blockwork/build/caching.py

```diff
@@ -48,17 +48,136 @@
 import hashlib
 import os
 from pathlib import Path
-from typing import Optional, TYPE_CHECKING
+import sys
+from types import ModuleType
+from typing import DefaultDict, Optional, TYPE_CHECKING
 if TYPE_CHECKING:
     from ..transforms import Transform
 from ..context import Context
+from ordered_set import OrderedSet as OSet
+import distutils.sysconfig
+import ast
+import site
+
 
 # Switch to never pull or push from the cache, but to instead compare the
 # hash with existing cache entries.
 # This will become a command option later...
 CACHE_CONSISTENCY_MODE = False
 
+class PyHasher:
+
+    def __init__(self):
+        self.module_stack: list[ModuleType] = []
+        self.dependency_map: DefaultDict[str, OSet[str]] = DefaultDict(OSet)
+        self.hash_map: dict[str, str] = {}
+        self.visitor = ast.NodeVisitor()
+        self.visitor.visit_Import = self.visit_Import
+        self.visitor.visit_ImportFrom = self.visit_ImportFrom
+
+        # Get a basic hash of the site
+        site_str =''
+        for sitepackages in site.getsitepackages():
+            site_str += ''.join(sorted(os.listdir(sitepackages)))
```

Comment:
Has this been through formatting? (I didn't think black did single quotes)

Reply:
Yes - but the pre-commit hook doesn't respect the rules from the pyproject.toml
We should unify these (I suspect there will be a lot of changes and would prefer to do that in its own PR)